service/history/api/reapplyevents/api.go

```diff
@@ -177,6 +177,7 @@ func Invoke(
 			reappliedEvents, err := eventsReapplier.ReapplyEvents(
 				ctx,
 				mutableState,
+				context.UpdateRegistry(ctx, mutableState),
```

Comment:
Please pass `nil` as a 2nd parameter everywhere. In all your cases, mutable state from context is not `nil`.

Reply:
Ah good catch! Thanks.